tlsn/tlsn-prover/src/tls/mod.rs

```diff
@@ -83,6 +83,26 @@ impl Prover<state::Initialized> {
             fut: Box::pin(async move { mux.run().await.map_err(ProverError::from) }.fuse()),
         };
 
+        // Sends configuration info to verifier for compatibility check
+        let mut configuration_fut = Box::pin({
+            let self_configuration = self.config.configuration_info.clone();
+            let mut mux_ctrl = mux_ctrl.clone();
+            async move {
+                let mut channel = mux_ctrl.get_channel("configuration").await?;
+                channel
+                    .send(TlsnMessage::ConfigurationInfo(self_configuration))
+                    .await?;
```

Comment:
@sinui0 I realised that if there's an error that occurs during configuration check, prover should receive the error back from verifier so that prover can log it for better debugging.

Initially I wanted to send `ConfigurationError` itself back from verifier, but the `source` trait object can't derive `Serialize` and `Deserialize` directly. There are crates like [typetag](https://docs.rs/typetag/0.2.16/typetag/) that might be able to help, but it doesn't seem straightforward. Meanwhile it kinda feels to me like the number of failure modes is kinda small in this case, hence I also considered reverting to exhaustive error enum.

Though before choosing either of the approaches, I wanted to get your thoughts in case there is a better way or something that I missed.